src/components/common/Interactive.tsx

```diff
@@ -108,36 +101,27 @@ const InteractiveBase = ({ onMove, onKey, ...rest }: Props) => {
         left: keyCode === 39 ? 0.05 : keyCode === 37 ? -0.05 : 0,
         top: keyCode === 40 ? 0.05 : keyCode === 38 ? -0.05 : 0,
       });
-    },
-    [onKeyCallback]
-  );
-
-  const handleMoveEnd = useCallback(() => setDragging(false), []);
+    };
 
-  const toggleDocumentEvents = useCallback(
-    (state) => {
+    function toggleDocumentEvents(state?: boolean) {
+      const touch = hasTouch.current;
       // add or remove additional pointer event listeners
-      const toggleEvent = state ? window.addEventListener : window.removeEventListener;
-      toggleEvent(hasTouched.current ? "touchmove" : "mousemove", handleMove);
-      toggleEvent(hasTouched.current ? "touchend" : "mouseup", handleMoveEnd);
-    },
-    [handleMove, handleMoveEnd]
-  );
+      const toggleEvent = state ? self.addEventListener : self.removeEventListener;
+      toggleEvent(touch ? "touchmove" : "mousemove", handleMove);
+      toggleEvent(touch ? "touchend" : "mouseup", handleMoveEnd);
+    }
 
-  useIsomorphicLayoutEffect(() => {
-    toggleDocumentEvents(isDragging);
-    return () => {
-      isDragging && toggleDocumentEvents(false);
-    };
-  }, [isDragging, toggleDocumentEvents]);
+    return [handleMoveStart, handleKeyDown, toggleDocumentEvents];
+  }, []);
+  useEffect(() => toggleDocumentEvents, []);
```

Comment:
We don't need `useIsomorphicLayoutEffect` as `useEffect` won't be called on the server-side

Also, we don't need the check, as in the case we try to unsubscribe from the function, which doesn't exist as a subscribed browser will do nothing.

Reply:
We use `useEffect` since Next.js shows tons of warnings about that =(